### PR TITLE
Reload module's sources when LS reloads buffer

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

LS needs to notify runtime that modules' sources need to be reloaded from FS, once its own buffer has been reloaded as well.

### Important Notes

Discovered during integration of https://github.com/enso-org/enso/pull/4050.
The test illustrates the problem if we don't reload module's sources - the sources essentially become stale even though they have changed.

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala [hunk 1]
```diff
@@ -2449,6 +2449,167 @@ class RuntimeServerTest
...
+
+    context.send(Api.Request(requestId, Api.PopContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.PopContextResponse(contextId))
+    )
+    // Push new item on the stack to trigger the re-execution
+    context.send(
+      Api.Request(
+        requestId,
+        Api.PushContextRequest(
```
Comment (4e6) [inline engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala]: btw, there is `Api.RecomputeContextRequest` if you just need to trigger the re-execution

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ReloadFileCmd.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  override def execute(implicit
+    ctx: RuntimeContext,
+    ec: ExecutionContext
+  ): Future[Unit] = {
+    ctx.locking.acquireFileLock(request.path)
+    try {
+      ctx.jobControlPlane.abortAllJobs()
+      ctx.jobProcessor.run(
+        new EnsureCompiledJob(Seq(request.path), forceCompilation = true)
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ReloadFileCmd.scala]: If you send module sources like `OpenFileCmd` does, and then use `ctx.executionService.setModuleSources` (which resets the compilation stage) then you don't need to change the compilation step adding `forceCompilation` flag.

File: engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala (hunk 1 above)
Comment (hubertp) [reply engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala]: Thanks!

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ReloadFileCmd.scala (hunk 2 above)
Comment (hubertp) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/ReloadFileCmd.scala]: applied

**Review by 4e6 — APPROVED** [review]

**Pull request merged**